src/components/AvatarCropModal/AvatarCropModal.js

```diff
@@ -0,0 +1,312 @@
+import PropTypes from 'prop-types';
+import React, {
+    useCallback, useEffect, useState,
+} from 'react';
+import {
+    Image, Pressable, SafeAreaView, View,
+} from 'react-native';
+import {GestureHandlerRootView} from 'react-native-gesture-handler';
+import {
+    interpolate,
+    useAnimatedGestureHandler,
+    useAnimatedStyle,
+    useSharedValue,
+    useWorkletCallback,
+} from 'react-native-reanimated';
+import CONST from '../../CONST';
+import compose from '../../libs/compose';
+import colors from '../../styles/colors';
+import styles from '../../styles/styles';
+import Button from '../Button';
+import HeaderWithCloseButton from '../HeaderWithCloseButton';
+import Icon from '../Icon';
+import * as Expensicons from '../Icon/Expensicons';
+import Modal from '../Modal';
+import Text from '../Text';
+import withLocalize, {withLocalizePropTypes} from '../withLocalize';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../withWindowDimensions';
+import ImageCropView from './components/ImageCropView';
+import Slider from './components/Slider';
+import imageManipulator from './libs/imageManipulator';
+
+const propTypes = {
+    /** Link to image for cropping   */
+    imageUri: PropTypes.string,
+
+    /** Callback to be called when user closes the modal */
+    onClose: PropTypes.func,
+
+    /** Callback to be called when user saves the image */
+    onSave: PropTypes.func,
+
+    ...withLocalizePropTypes,
+    ...windowDimensionsPropTypes,
+};
+
+const defaultProps = {
+    imageUri: null,
+    onClose: () => { },
+    onSave: () => { },
+};
+
+// This component can't be written using class since reanimated API uses hooks.
+const AvatarCropModal = (props) => {
+    const originalImageWidth = useSharedValue(CONST.AVATAR_CROP_MODAL.INITIAL_SIZE);
+    const originalImageHeight = useSharedValue(CONST.AVATAR_CROP_MODAL.INITIAL_SIZE);
+    const translateY = useSharedValue(0);
+    const translateX = useSharedValue(0);
+    const scale = useSharedValue(CONST.AVATAR_CROP_MODAL.MIN_SCALE);
+    const rotation = useSharedValue(0);
+    const translateSlider = useSharedValue(0);
+
+    const [imageContainerSize, setImageContainerSize] = useState(CONST.AVATAR_CROP_MODAL.INITIAL_SIZE);
+    const [sliderLineWidth, setSliderContainerSize] = useState(CONST.AVATAR_CROP_MODAL.INITIAL_SIZE);
+
+    // An onLayout callback, that initializes the image container, for proper render of an image
```

Comment:
not a blocker: This comment could be improved by explaining "why". Currently, it just explains _what the code is doing_.